myploty.py

```diff
@@ -1,5 +1,189 @@
+            go.Scatter3d(
+                x=[point[0], arrow_end[0]],
+                y=[point[1], arrow_end[1]],
+                z=[point[2], arrow_end[2]],
+                mode="lines",
+                line=dict(color="red", width=2),
+                hoverinfo="none",
+                showlegend=False,
+            )
+        )
+
+    fig.update_layout(
+        # title=f'Position: {position.split(".txt")[0]} - Similarity: {similarity:.2f}',
+        scene=dict(
+            xaxis_title="Longitude",
+            yaxis_title="Latitude",
+            zaxis_title="Altitude",
+            aspectmode="data",
+            camera=dict(eye=dict(x=1.25, y=1.25, z=1.25)),
+        ),
+        title="Point Cloud with Normals",
+    )
+
+    fig.show()
+
+
+def plot_cloud_with_trace_plotly(point_cloud):
+    fig = go.Figure()
+
+    # Plotting the points
+    fig.add_trace(
+        go.Scatter3d(
+            x=point_cloud[:, 0],
+            y=point_cloud[:, 1],
+            z=point_cloud[:, 2],
+            mode="markers",
+            marker=dict(color="blue", size=5),
+            name="Point Cloud",
+        )
+    )
+
+    # Plotting the normals as arrows
+    fig.add_trace(
+        go.Scatter3d(
+            x=point_cloud[:, 0],
+            y=point_cloud[:, 1],
+            z=point_cloud[:, 2],
+            mode="lines",
+            line=dict(color="red", width=2),
+            hoverinfo="none",
+            showlegend=False,
+        )
+    )
+
+    fig.update_layout(
+        scene=dict(
+            xaxis_title="X",
+            yaxis_title="Y",
+            zaxis_title="Z",
+            aspectmode="data",
+            camera=dict(eye=dict(x=1.25, y=1.25, z=1.25)),
+        ),
+        title="Point Cloud with Normals",
+    )
+
+    fig.show()
+
+
+def plot_cloud_with_normals(point_cloud, normals):
+    """
+    Plot the point cloud with normals.
+    :param point_cloud: Nx3 numpy array of points.
+    :param normals: Nx3 numpy array of corresponding normals.
+    """
+    fig = plt.figure(figsize=(8, 6))
+    ax = fig.add_subplot(111, projection="3d")
+
+    # Plotting the points
+    ax.scatter(
+        point_cloud[:, 0],
+        point_cloud[:, 1],
+        point_cloud[:, 2],
+        color="b",
+        label="Point Cloud",
+    )
+
+    # Plotting the normals
+    for i in range(len(point_cloud)):
+        if not np.isnan(normals[i]).any():  # Check if normals were computed
+            ax.quiver(
+                point_cloud[i, 0],
+                point_cloud[i, 1],
+                point_cloud[i, 2],
+                normals[i, 0],
+                normals[i, 1],
+                normals[i, 2],
+                length=0.2,
+                color="r",
+                normalize=True,
+            )
+
+    ax.set_xlabel("X")
+    ax.set_ylabel("Y")
+    ax.set_zlabel("Z")
+    ax.legend()
+    ax.set_title("Point Cloud with Normals")
+    plt.show()
 
 
 def plot_point_cloud_plotly(position: str, similarity, cloud_data):
```

Comment:
Filtering the point cloud in the plot functionality is not suitable. Thin out the dataset at first and then plot them.